src/test/java/uk/gov/hmcts/darts/task/service/impl/AutomatedTasksMapperTest.java

```diff
@@ -61,5 +62,67 @@ void shouldMapEntityToDetailedAutomatedTask() {
         assertThat(result.getLastModifiedBy()).isEqualTo(2);
         assertThat(result.getIsCronEditable()).isTrue();
         assertThat(result.getBatchSize()).isEqualTo(50);
+        assertThat(result.getRpoCsvEndHour()).isNull();
+        assertThat(result.getRpoCsvStartHour()).isNull();
+        assertThat(result.getArmReplayStartTs()).isNull();
+        assertThat(result.getArmReplayEndTs()).isNull();
+        assertThat(result.getArmAttributeType()).isNull();
+    }
+
+    @Test
+    void shouldMapEntityToDetailedAutomatedTaskWithArmTask() {
+        // Given
+        UserAccountEntity createdBy = new UserAccountEntity();
+        createdBy.setId(1);
+
+        UserAccountEntity lastModifiedBy = new UserAccountEntity();
+        lastModifiedBy.setId(2);
+
+        OffsetDateTime createdAt = OffsetDateTime.of(2023, 10, 1, 10, 0, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime lastModifiedAt = OffsetDateTime.of(2023, 10, 2, 12, 0, 0, 0, ZoneOffset.UTC);
+
+        AutomatedTaskEntity entity = new AutomatedTaskEntity();
+        entity.setId(123);
+        entity.setTaskName("Test Task");
+        entity.setTaskEnabled(true);
+        entity.setTaskDescription("A test task for demonstration");
+        entity.setCronExpression("0 0 * * *");
+        entity.setCreatedDateTime(createdAt);
+        entity.setCreatedBy(createdBy);
+        entity.setLastModifiedDateTime(lastModifiedAt);
+        entity.setLastModifiedBy(lastModifiedBy);
+        entity.setCronEditable(true);
+        entity.setBatchSize(50);
```

Comment:
I think these lines are the same in both tests, it is worth pulling into a private method to prevent the duplication?

Reply:
agree - have refactored